src/ape_ethereum/multicall/handlers.py

```diff
@@ -0,0 +1,290 @@
+from typing import Any, Iterator, List, Tuple, Union
+
+from ape.api import ReceiptAPI, TransactionAPI
+from ape.contracts.base import (
+    ContractCallHandler,
+    ContractInstance,
+    ContractMethodHandler,
+    ContractTransactionHandler,
+    _select_method_abi,
+)
+from ape.exceptions import ChainError
+from ape.types import ContractType, HexBytes
+from ape.utils import ManagerAccessMixin, cached_property
+from ape.utils.abi import MethodABI
+
+from .constants import (
+    MULTICALL3_ADDRESS,
+    MULTICALL3_CODE,
+    MULTICALL3_CONTRACT_TYPE,
+    SUPPORTED_CHAINS,
+)
+from .exceptions import InvalidOption, NotExecutedError, UnsupportedChainError, ValueRequired
+
+
+class BaseMulticall(ManagerAccessMixin):
+    def __init__(self) -> None:
+        """
+        Initialize a new Multicall session object. By default, there are no calls to make.
+        """
+        self.calls: List[dict] = []
+
+    @classmethod
+    def deploy(cls):
```

Comment:
Thinking of renaming this to `inject`

Reply:
[d00958c](https://github.com/ApeWorX/ape/pull/1189/commits/d00958cfa15fb76e2d9ea515a1b6e570219c2f6d)